Library/CorrelationIdManager.ts

```diff
@@ -65,7 +65,7 @@ class CorrelationIdManager {
                         }
                         delete CorrelationIdManager.pendingLookups[appIdUrlString];
                     });
-                } else if (res.statusCode >= 400 && res.statusCode < 500) {
+                } else if (res.statusCode >= 400 && res.statusCode < 500 || res.statusCode === undefined) {

```

Comment:
Can undefined show up for reasons that _should_ retry? Like an abrupt disconnection

Reply:
I'm going based off of this
https://stackoverflow.com/questions/29752225/http-statuscode-sometimes-undefined
I haven't been able to repro this outside of unit tests, but I'm still hesitant of it because it seems like undefined behavior